src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/gadgets/MultiTool.java

```diff
@@ -86,12 +94,33 @@ protected ItemUseHandler getItemUseHandler() {
                     }
                 }
             } else {
+                final String LORE_PREFIX = ChatColors.color("&8\u21E8 &7Mode: ");
+                final Pattern REGEX = Pattern.compile(ChatColors.color("(&c&o)?" + LORE_PREFIX) + "(.+)");
                 index = nextIndex(index);
 
                 SlimefunItem selectedItem = modes.get(index).getItem();
                 String itemName = selectedItem != null ? selectedItem.getItemName() : "Unknown";
                 Slimefun.getLocalization().sendMessage(p, "messages.multi-tool.mode-change", true, msg -> msg.replace("%device%", "Multi Tool").replace("%mode%", ChatColor.stripColor(itemName)));
-                selectedMode.put(p.getUniqueId(), index);
+                
+                ItemMeta meta = item.getItemMeta();
+                List<String> lore = meta.hasLore() ? meta.getLore() : new ArrayList<>();
+                
+                meta.getPersistentDataContainer().set(key, PersistentDataType.INTEGER, index);
```

Comment:
Let's not rely on indexes, it's bad we do it currently let's not expand that. Let's instead just set the MultiToolMode here

Reply:
I tried to replace indexes with item IDs, hopefully this is what you meant.